components/gitpod-db/src/typeorm/migration/1612781029090-UserStorageUserIdChar36.ts

```diff
@@ -0,0 +1,12 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class UserStorageUserIdChar1612781029090 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+      await queryRunner.query("ALTER TABLE d_b_user_storage_resource MODIFY userId char(36);");
```

Comment:
is it a problem?
changing `userId varchar(255)` to fixed `char(36)` sound like there is some risk.

Reply:
> is it a problem?

Yes, it is for the lookup performance. The DB cannot assume that the 219 characters that are always blank. Ideally they all end up in one leaf of the Red-Black-tree but one never know. Also, it blows up the memory used up by the index.

> changing userId varchar(255) to fixed char(36) sound like there is some risk.

There is no risk to break something as we never used anything else than UUIDs (36 characters), and we're maintaining that limit with the current id change.
During deployment however this might be a problem as it might take some time because it requires re-creating indexes.